department-of-veterans-affairs/caseflow
# Kamalam7/appeals 35568

Resolves https://jira.devops.va.gov/browse/APPEALS-35568

# Description
In the Edit CMP Information Modal, VADOR date is enable for the Mail Supervisor and they change or modify the date and once they click on Save button it will update the CMP Information table in the Review package page.
When the general mail users won't have access to change or modify the VADOR, where it will be disabled.

## Acceptance Criteria
- [x] The mail supervisor is able to edit the VA DOR on the "Edit CMP information" modal
       After clicking the "save" button, the VA DOR is updated
       The regular mail user is unable to edit the VA DOR on the the "Edit CMP information" modal
       The VA DOR is grayed out on the modal

## Testing Plan
1. Login as a Mail user.
2. Click on Queue.
3. In the Switch views, select Correspondence cases.
4. Click on any Veteran, it will take to the Review package page.
5. In the Review package page, beside CMP Information you can see the Edit button, click on the Edit. A modal popups.
6. In the Edit modal, when you Login as a regular mail user, the VADOR field is disabled.
<img width="834" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110078646/270d209b-57fd-4ebf-848b-1aa2f184789b">
7. When you Login like a Mail Supervisor, the VADOR date is enabled and you can change the date and save it. 
<img width="848" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110078646/565955cd-8fa3-4713-90d5-c287d32113ba">
8. Once the Mail Supervisor changes the VADOR and save, it will update the VADOR in the CMP information table.



# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
